### ISSUE TITLE
Exception "Attempt to decode JSON with unexpected mimetype: text/plain; charset=utf-8" thrown during startup

### ISSUE THREAD

**Issue by robinsmidsrod** [gjohansson-ST/sector#12]
When I start Home Assistant I get this message from the Sector custom component (installed via HACS), and all entities are unavailable.

I noticed that the web UI and the app is also unable to login, so I think there might be an outage on Sector's side, but I think the component should handle this exception somewhat more gracefully, hopefully with a delayed setup and an exponential backoff retry timer.

Sector component version: 0.1.9

```
2020-08-15 18:01:42 ERROR (MainThread) [homeassistant.setup] Error during setup of component sector
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 191, in _async_setup_component
    result = await task
  File "/config/custom_components/sector/__init__.py", line 77, in async_setup
    data_out = await response.json()
  File "/usr/local/lib/python3.8/site-packages/aiohttp/client_reqrep.py", line 1026, in json
    raise ContentTypeError(
aiohttp.client_exceptions.ContentTypeError: 0, message='Attempt to decode JSON with unexpected mimetype: text/plain; charset=utf-8', url=URL('https://mypagesapi.sectoralarm.net/api/Login/Login')
```

**Comment by gjohansson-ST** [comment]
Implemented CannotConnect. v0.2.0-beta1

**Comment by robinsmidsrod** [comment]
@gjohansson-ST Does this mean you've fixed the issue? The message seems to indicate you've fixed it, but I'm not sure where to look for confirmation. If you have fixed it, when to you expect a release to HACS?

**Comment by gjohansson-ST** [comment]
> @gjohansson-ST Does this mean you've fixed the issue? The message seems to indicate you've fixed it, but I'm not sure where to look for confirmation. If you have fixed it, when to you expect a release to HACS?

I have released a beta1 so if you in HACS turn on beta-release for Sector then you find it and can try it out.

**Comment by robinsmidsrod** [comment]
> In HACS turn on beta-release for Sector.

I might be stupid, but I don't see any settings in the "Community" section in Home Assistant to enable beta releases. Got a pointer to some documentation on that?

**Comment by gjohansson-ST** [comment]
Sorry for bad explanation.
Click on the three dots next to Sector integration within HACS.
Choose reinstall
Click to turn on "Show beta versions"
Choose to install the newest one released "v0.2.0-beta1"

**Comment by robinsmidsrod** [comment]
Thanks a bunch! Very helpful! Wasn't aware that this was the way to use beta releases in HACS. Got the 0.2.0-beta1 up and it seems to work. Tried to turn my alarm on and off and that seems to work. Let's hope it works better the next time they have an outage. I guess it can be simulated with a bit of network trickery. Have you already done this testing on your own?

**Comment by gjohansson-ST** [comment]
Not yet, therefore just a beta so far.
Nice if you also test it as you have it installed now and report back.

Thanks

**Comment by robinsmidsrod** [comment]
@gjohansson-ST Actually, my comment back then that testing would be somewhat easy was premature. Since Sector's pages are HTTPS, I'd have to do a man-in-the-middle attack to make it show a page in text/plain instead of application/json. Not sure how to do this, so I don't know how to test the problem I reported.

**Comment by gjohansson-ST** [comment]
> @gjohansson-ST Actually, my comment back then that testing would be somewhat easy was premature. Since Sector's pages are HTTPS, I'd have to do a man-in-the-middle attack to make it show a page in text/plain instead of application/json. Not sure how to do this, so I don't know how to test the problem I reported.

Good point. Did not consider that. Was purely thinking on if internet was working or not. Let me see if I can test it somehow.

**Comment by gjohansson-ST** [comment]
Closed with v0.3.0
